src/data/ab-attrs/ab-attr.ts

```diff
@@ -1,25 +1,28 @@
 import type { AbAttrCondition } from "#app/@types/AbAttrCondition";
 import { type Pokemon } from "#app/field/pokemon";
+import type { Ability } from "#app/data/ability";
 
 export abstract class AbAttr {
+  public source: Ability;
   public showAbility: boolean;
+  public showAbilityInstant: boolean;
   private extraCondition: AbAttrCondition;
 
-  constructor(showAbility: boolean = true) {
+  constructor(showAbility: boolean = true, showAbilityInstant: boolean = false) {
     this.showAbility = showAbility;
+    this.showAbilityInstant = showAbilityInstant;
   }
 
   /**
    * Applies the effects of this attribute
-   * @param _pokemon The {@linkcode Pokemon} with the ability
-   * @param _passive `true` if the source ability is a Pokemon's passive
-   * @param _simulated `true` if attribute effects should be resolved without changing game state
-   * @param _args Any additional parameters or data to modify
+   * @param pokemon The {@linkcode Pokemon} with the ability
+   * @param simulated `true` if attribute effects should be resolved without changing game state
+   * @param args Any additional parameters or data to modify
```

Comment:
```suggestion
   * @param _pokemon The {@linkcode Pokemon} with the ability
   * @param _simulated `true` if attribute effects should be resolved without changing game state
   * @param _args Any additional parameters or data to modify
```

They have to match the names, don't they?

Reply:
It's not a strict rule in the sense that the params are still highlighted and the links are still functional

Removing the underscores makes the docs a bit neater for subclasses as their `apply` overrides are being made, which is the main point of this and similar comment blocks.